File: lib/features/activity_hiking/models/hiking_conditions_report.dart
import 'package:turbo/features/activity_fishing/api.dart' show WeatherSlice;

/// Server-derived hiking conditions report. Reuses the shared
/// [WeatherSlice] parser from the fishing kind (same wire shape; one
/// Dart class for all kinds).
class HikingConditionsReport {
  final String activityId;
  final DateTime validAt;
  final DateTime fetchedAt;
  final WeatherSlice weather;
  final int? score;
  final String rationale;

  const HikingConditionsReport({
    required this.activityId,
    required this.validAt,
    required this.fetchedAt,
    required this.weather,
    required this.score,
    required this.rationale,
  });

  factory HikingConditionsReport.fromJson(Map<String, dynamic> json) =>
      HikingConditionsReport(
        activityId: json['activityId'] as String,
        validAt: DateTime.parse(json['validAt'] as String),
        fetchedAt: DateTime.parse(json['fetchedAt'] as String),
        weather: WeatherSlice.fromJson(json['weather'] as Map<String, dynamic>),
        score: (json['score'] as num?)?.toInt(),
        rationale: json['rationale'] as String,
      );
}

File: apps/api/src/Activities/Turbo.Activities.Hiking.Contracts/value/HikingConditionsReport.cs
using System.Text.Json.Serialization;
using Turboapi.Activities.value;

namespace Turboapi.Activities.Hiking.value;

public sealed record HikingConditionsReport
{
    [JsonPropertyName("activityId")] public Guid ActivityId { get; init; }
    [JsonPropertyName("validAt")] public DateTimeOffset ValidAt { get; init; }
    [JsonPropertyName("fetchedAt")] public DateTimeOffset FetchedAt { get; init; }
    [JsonPropertyName("weather")] public WeatherSlice Weather { get; init; }
    [JsonPropertyName("score")] public int? Score { get; init; }
    [JsonPropertyName("rationale")] public string Rationale { get; init; }

    [JsonConstructor]
    public HikingConditionsReport(
        Guid activityId, DateTimeOffset validAt, DateTimeOffset fetchedAt,
        WeatherSlice weather, int? score, string rationale)
    {
        ActivityId = activityId;
        ValidAt = validAt;
        FetchedAt = fetchedAt;
        Weather = weather;
        Score = score;
        Rationale = rationale ?? throw new ArgumentNullException(nameof(rationale));
    }
}
